subprojects/packagefiles/pcre2/meson.build

```diff
@@ -6,25 +6,76 @@ project(
   version: '10.42',
 )
 
-c_compiler = meson.get_compiler('c')
 
-pcre2_h = configure_file(
-  input: 'src/pcre2.h.generic',
-  output: 'pcre2.h',
-  copy: true,
-)
+platform_is_dragonflybsd = (host_machine.system() == 'dragonfly')
+platform_is_freebsd = (host_machine.system() == 'freebsd')
+platform_is_netbsd = (host_machine.system() == 'netbsd')
+platform_is_openbsd = (host_machine.system() == 'openbsd')
+
+platform_is_bsd = (
+    platform_is_dragonflybsd or
+    platform_is_freebsd or
+    platform_is_netbsd or
+    platform_is_openbsd or
+false)
+
+platform_is_emscripten = (host_machine.system() == 'emscripten')
+
+platform_is_haiku = (host_machine.system() == 'haiku')
+
+platform_is_linux_desktop = (host_machine.system() == 'linux')
+platform_is_android = (host_machine.system() == 'android')
+platform_is_linux = platform_is_android or platform_is_linux_desktop
+
+platform_is_darwin = (host_machine.system() == 'darwin')
+
+platform_is_cygwin = (host_machine.system() == 'cygwin')
+
+platform_is_unixlike = (
+    platform_is_bsd or
+    platform_is_emscripten or
+    platform_is_haiku or
+    platform_is_linux or
+    platform_is_darwin or
+    platform_is_cygwin or
+false)
```

Comment:
typo?

Reply:
is this in regard to cygwin?
This is a straight translation from the upstream `CMakeLists.txt` that uses `UNIX` which is true on cygwin.

Reply:
The last false

Reply:
This was originally taken from SDL https://github.com/mesonbuild/wrapdb/blob/4f0620ca571968a58113f6e26126274de99eca91/subprojects/packagefiles/sdl2/meson.build#L71-L76
but there seems to be no good reason for it to be there.

Appears to be something we inherited from the taisei SDL2 port